src/btf.rs

```diff
@@ -155,6 +155,9 @@ impl Btf {
     /// Resolve a name referenced by a Type which is defined in the current BTF
     /// object.
     pub fn resolve_name<T: BtfType + ?Sized>(&self, r#type: &T) -> Result<String> {
+        if r#type.get_name_offset()? == 0 && !r#type.can_be_anon() {
+            return Err(Error::InvalidString(0));
+        }
```

Comment:
IMO we should not special case id 0 at this level. This is already handled at the obj level, which helps keeping things handled at the right level (`Btf` mostly handles the split vs base logic) and helps with maintenance. Also the shortcut won't save much here.